flymake-eslint.el

```diff
@@ -261,9 +261,7 @@ argument."
   (let ((default-directory
          (or
           flymake-eslint-project-root
-          (when (and (featurep 'project)
-                     (project-current))
-            (project-root (project-current)))
```

Comment:
is there a reason to remove `(project-root (project-current))` altogether, rather than keeping it as a fallback?

Reply:
I'm fine keeping the current code as fallback. I updated my patch.

Out of curiosity, do you have any use case in mind where `project-root` would return the right result but `locate-dominating-file` wouldn't?

Reply:
wouldn't it be needed for users who have a `.git` or similar but no `package.json`?  for instance, at my job we have a tooling repo containing some zero-dependency scripts in various languages (bash, node.js, etc).

Reply:
I never thought of that, you are right. Then, what about searching for an eslint configuration file instead of package.json? It seams there must be one for eslint to work.

Reply:
I think that's a great idea :+1:

Reply:
To get highest chance of success (see commit message for details), the end result is a little more complicated than that. What do you think?